client/live_test.go

```diff
@@ -48,6 +48,16 @@ func (s *LiveTests) TestAuthenticateFail(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, "authentication failed(\n|.)*")
 }
 
+func (s *LiveTests) TestNoUnauthorisedAuthenticateFail(c *gc.C) {
+	cred := *s.cred
+	cred.Region = ""
+	osclient := client.NewClient(&cred, s.authMode, nil)
+	c.Assert(osclient.IsAuthenticated(), gc.Equals, false)
+	err := osclient.Authenticate()
+	isAuthorised := gooseerrors.IsUnauthorised(err)
```

Comment:
This is confusing. You are asking if an error is unauthorised, and then assigning that value into isAuthorised.

Are we expecting a credential without a region to return an unauthorised error?